src/components/site-form.tsx

```diff
@@ -480,9 +492,34 @@ export const SiteForm = ( {
 											</div>
 										) }
 
-										{ setUseCustomDomain && setCustomDomain && (
+										{ useCustomDomain && setEnableHttps && (
+											<div className="flex items-center gap-2 mt-4">
+												<input
+													type="checkbox"
+													id="enable-https"
+													checked={ enableHttps }
+													onChange={ ( e ) => setEnableHttps( e.target.checked ) }
+												/>
+												<label htmlFor="enable-https">{ __( 'Enable HTTPS' ) }</label>
+											</div>
+										) }
+
+										{ useCustomDomain && setEnableHttps && (
 											<div className="text-a8c-gray-50 text-xs mt-2">
-												{ __( 'Your system password will be required to set up the domain.' ) }
+												{ __(
+													'You need to manually add the Studio certificate authority to your keychain and trust it.'
+												) }{ ' ' }
```

Comment:
```suggestion
												{ __(
													'You need to manually add the Studio root certificate authority to your keychain and trust it to enable HTTPS.'
												) }{ ' ' }
```
"Root certificate authority" is what it's called in the MacOS Keychain Access app. Adding `to enable HTTPS` to the end also clarifies that this instruction is only relevant if the checkbox is ticked.

Aside from tweaking the wording, let's also hide this on Windows.